### PR TITLE
tweak ephemeral error messages

### PR DESCRIPTION
**Pull request by joshblum**
patch does the following:

- applies supersedes to ERROR type messages so erred exploding messages can have proper explode now info.
- hides error messages on already exploded messages but not ash lines
- tweaks error message copy/display

image

cc @cecileboucheron let me know if you want any other copy tweaks on the error text.

### CODE REVIEW

**Comment by joshblum** [comment]
(ci flakes unrelated)

File: protocol/avdl/chat1/local.avdl [hunk 1]
```diff
@@ -511,7 +511,7 @@ protocol local {
     MessageType messageType;
     gregor1.Time ctime;
     boolean isEphemeral;
-    boolean isEphemeralExpired;
+    union { null, string } explodedBy;
```
Comment (songgao) [inline protocol/avdl/chat1/local.avdl]: Can we have both? This way JS side don't need to do the comparison.

File: shared/chat/conversation/messages/wrapper/index.tsx [hunk 2]
```diff
@@ -250,7 +250,10 @@ class _WrapperMessage extends React.Component<Props & Kb.OverlayParentProps, Sta
   _isFailed = () =>
     !!this.props.failureDescription && (
       <Kb.Text key="isFailed" type="BodySmall">
-        <Kb.Text type="BodySmall" style={styles.fail}>
+        <Kb.Text type="BodySmall" style={this._isExploding() ? styles.failExploding : styles.fail}>
+          {this._isExploding() && (
+            <Kb.Icon fontSize={16} boxStyle={styles.failExplodingIcon} type="iconfont-block" />
+          )}{' '}
```
Comment (songgao) [inline shared/chat/conversation/messages/wrapper/index.tsx]: Does this insert a space when `this._isExploding() === false`?

File: protocol/avdl/chat1/local.avdl (hunk 1 above)
Comment (joshblum) [reply protocol/avdl/chat1/local.avdl]: We'd have to have a derived type that computes the value before sending it to the frontend. To keep things simple I just let the frontent derive it, we do a similar calculation here https://github.com/keybase/client/pull/21876/files#diff-ff27bea5c579bc43b5b22d046db6acc5R1214

File: shared/chat/conversation/messages/wrapper/index.tsx (hunk 2 above)
Comment (joshblum) [reply shared/chat/conversation/messages/wrapper/index.tsx]: used some padding instead!

**Comment by songgao** [comment]
I glanced over the Go side, and it LGTM too. But I'm not super familiar with this code, so feel free to wait for Mike.

**Comment by joshblum** [comment]
> I'm OK with this, but it seems space would be more consistent. Does this not work?
> 
> ```
>           {this._isExploding() && (
>             <>
>               <Kb.Icon fontSize={16} boxStyle={styles.failExplodingIcon} type="iconfont-block" />
>               {' '}
>             </>
>           )}
> ```

Oh great, I wasn't sure the syntax, did not know about  `<>...</>`, thanks!

File: go/chat/supersedes.go [hunk 3]
```diff
@@ -132,21 +155,24 @@ func (t *basicSupersedesTransform) transformUnfurl(msg chat1.MessageUnboxed, sup
 }
 
 func (t *basicSupersedesTransform) transform(ctx context.Context, msg chat1.MessageUnboxed,
-	superMsgs []chat1.MessageUnboxed) *chat1.MessageUnboxed {
+	superMsgs []chat1.MessageUnboxed) (*chat1.MessageUnboxed, bool) {
```
Comment (mmaxim) [inline go/chat/supersedes.go]: Name these return variables so we can know what this bool is.

**Pull request merged**